prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -316,7 +316,22 @@ void rxr_pkt_data_from_tx(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entry,
 	desc = tx_entry->desc[0];
 	assert(tx_iov_index < tx_entry->iov_count);
 	assert(tx_iov_offset < tx_entry->iov[tx_iov_index].iov_len);
-	if (!tx_entry->desc[tx_iov_index]) {
+
+	/*
+	 * We want to go through the bounce-buffers when one of these three
+	 * conditions true:
+	 * - Applications can not honor FI_MR_LOCAL
+	 * - desc.peer.iface is anything but FI_HMEM_SYSTEM
+	 * - This is for a intra-node transfer via shared-memory
+	 *
+	 * In the first case, we use the pre-registered pkt_entry's MR. In the
+	 * second case, this is for the eager and medium-message protocols which
+	 * can not rendezvous and pull the data from a peer. In the third case,
+	 * the bufpool would not have been created with a registration handler,
+	 * so pkt_entry->mr will be NULL.
+	 *
+	 */
+	if (!tx_entry->desc[tx_iov_index] && pkt_entry->mr) {
```

Comment:
I think this change need to be in a separate commit.

Reply:
I am a little confused. The comment and implementation is that we will use bounce buffer for shm provider, but the title of the commit is that we do not use bounce buffer for shm provider.

Reply:
Title and the code are right, the last bullet in my comment is flipped. Will fix it, thanks!